sashimi-webapp/src/logic/filemanager/file.js

```diff
@@ -0,0 +1,133 @@
+import storage from '../../database/storage';
+import Folder from './folder';
+
+/**
+* File Object
+*
+* @param {Number} fileID
+* @param {String} fileName
+* @param {String} filePath
+*/
+
+
+export default function File(fileID, fileName, filePath, parentFolder) {
+  this.id = fileID;
+  this.name = fileName;
+  this.path = filePath;
+  this.parentFolder = parentFolder;
+  console.log('new File created');
+}
+
+/**
+ * Remove file from the database
+ *
+ * @param {}
+ * @return {}
+ */
+File.prototype.remove = function remove() {
+  console.log('file.remove');
+  // Case 1a: file exist, file removed
+  // Case 1b: file exist, file not removed (error)
+  // Case 2: file does not exist (nothing removed)
+
+  return storage.deleteFile(this.path)
+    .then(() => {
+      const parentFolder = this.parentFolder;
+      const index = parentFolder.childFileList.findIndex(childFile => childFile.id === this.id);
+      parentFolder.childFileList.splice(index, 1);
+    })
+    .catch((error) => {
+      throw error;
+    });
+};
+
+/**
+ * Save data into file
+ *
+ * @param {String} data
+ * @return {}
+ */
+File.prototype.save = function save(data) {
+  console.log('file.save');
+  // Case 1a: file exist, file saved
+  // Case 1b: file exist, file not saved (error)
+  // Case 2: file does not exist (nothing saved)
+
+  return storage.saveFile(this.id, data);
```

Comment:
The intended behaviour of this code was to simply carry the promise from `storage` to the `fileManager` UI. 

The error isn't caught here because it wasn't intended to be handled here. Instead the `fileManager` UI should be handling the error instead. (e.g. To display "failed to save file") 

Aside from that, 

```
storage.saveFile(this.id, data)
.then(() => true)
.catch(err => false);
```

If `storage.saveFile` failed to run (let's say it has called `reject` in its promise), the error will get consumes by this line of code `catch(err => false);`. This will result in `fileManager` UI not being able to receive the error. 

Let's say if we ammend the code to catch the error and propagate the error to `fileManager`, like this, 
 ```
storage.saveFile(this.id, data)
.then(() => true)
.catch(err => throw err);
```

Then, it will be as good as not catching the error, since it is not going to be handled here. 

---

`.then(() => true)` is not needed too. 
In this current implement, `FileManager ` can know whether the code run correctly or not by following this rule where:
- `.then` will be executed on success
- `.catch` will be called on failure

E.g.

```
File.save()
.then(() => console.log('file saved successfully')
.catch((err) => console.log('Uh no, I cannot save the file. Here is the error ' + err.message));

```